web_advanced_search/static/src/js/web_advanced_search.js

```diff
@@ -283,7 +284,9 @@ odoo.define("web_advanced_search", function(require) {
                 return FieldManagerMixin._applyChanges.apply(this, arguments);
             }
 
-            return $.Deferred().resolve();
+            return new Promise(resolve => {
+                resolve();
+            });
```

Comment:
I think this is not the same.

We were returning a resolved deferred before, so it should be something like:

```suggestion
            return new Promise().resolve();
```